lib/hawkular/inventory/entities.rb

```diff
@@ -0,0 +1,217 @@
+# It contains class definitions that are used by the inventory REST client
+module Hawkular::Inventory
+  # A Basic inventory entity with id, name, path and optional properties
+  class BaseEntity
+    # @return [String] Full path of the entity
+    attr_reader :path
+    # @return [String] Name of the entity
+    attr_reader :name
+    # @return [String] Name of the entity
+    attr_reader :id
+    # @return [String] Feed this entity belongs to (or nil in case of a feedless entity)
+    attr_reader :feed
+    # @return [String] Name of the environment for this entity
+    attr_reader :env
+    # @return [String] Properties of this entity
+    attr_reader :properties
+
+    def initialize(hash)
+      @id = hash['id']
+      @path = hash['path']
+      @name = hash['name'] || @id
+      @properties = hash['properties'] || {}
+      @_hash = hash.dup
+
+      return if @path.nil?
+
+      tmp = @path.split('/')
+      tmp.each do |pair|
+        (key, val) = pair.split(';')
+        case key
+        when 'f'
+          @feed = val
+        when 'e'
+          @env = val
+        end
+      end
+    end
+
+    def ==(other)
```

Comment:
OK so you can fix it here or in a separate PR if this is not a new code. your call.
And `equal?` vs `eql?` is definitely confusing especially coming from Java...